test/cqlpy/test_service_level_api.py

```diff
@@ -53,22 +59,18 @@ def wait_until_all_connections_authenticated(cql, wait_s = 1, timeout_s = 30):
     
     raise RuntimeError(f"Awaiting for connections authentication timed out.")
 
-def wait_for_scheduling_group_assignment(cql, user, scheduling_group, wait_s = 2, timeout_s = 60):
-    start_time = time.time()
-    while time.time() - start_time < timeout_s:
-        connections = cql.execute(f"SELECT username, scheduling_group FROM system.clients WHERE client_type='cql' AND username='{user}' ALLOW FILTERING")
-        
-        require_wait = False
-        for row in connections:
-            if row[1] != f"sl:{scheduling_group}":
-                require_wait = True
-                break
-        if require_wait:
-            time.sleep(wait_s)
-            continue
-        return
+# The driver creates 1 connection per shard plus 1 control connection.
```

Comment:
This probably means that this test is only valid with Scylla's driver, not with Cassandra's driver :-( That's not great because our intention is to test Scylla - not Scylla's driver. But I'll let it slide. The test will work as expected on CI and developers who are annoyed that it fails (if it does fail) on Cassandra's driver can send a patch later.

Reply:
I don't insist here, but just for general information:

The issue of which **driver** you have installed, the one from Cassandra (actually Datastax) or the one from Scylla, is separate for which database you use, Scylla and Cassandra. The Cassandra driver works well against Scylla (it needs to, we need to be compatible), and you can run the tests with it. But as I said, I won't insist, nobody excepts me ever runs Scylla's tests with Cassandra's drivers (I sometimes do, just in spite ;-)).

Reply:
Perhaps we can adjust the check. We can either:

- Just check that at least n-1 out of n connections have been adjusted to use the new scheduling group,
- Figure out how to force the python driver to send some request on the control connection, then adjust the check to verify that all connections have the new scheduling group.

Both of these will work with the upstream driver.